rama-net/src/client/pool.rs

```diff
@@ -213,20 +230,30 @@ where
         );
 
         let mut storage = self.storage.lock();
-        let pooled_conn = {
-            storage
-                .iter()
-                .position(|stored| &stored.id == id)
-                .and_then(|idx| storage.remove(idx))
-        };
 
-        if let Some(pooled_conn) = pooled_conn {
-            return Ok(ConnectionResult::Connection(LeasedConnection {
-                active_slot,
-                pooled_conn: Some(pooled_conn),
-                returner: self.returner.clone(),
-                failed: false.into(),
-            }));
+        let idx = storage.iter().position(|stored| &stored.id == id);
+        let pooled_conn = idx.and_then(|idx| storage.remove(idx));
+
+        if let (Some(idx), Some(pooled_conn)) = (idx, pooled_conn) {
+            let mut timeout_triggered = false;
+            if let Some(timeout) = self.idle_timeout {
+                if pooled_conn.last_used.elapsed() > timeout {
+                    timeout_triggered = true;
+                }
+            }
+
+            // Since we always insert in the beginning we can assume if this connection has been idle for too long
+            // all connections older then this one should also be evicted
+            if timeout_triggered {
```

Comment:
can you create a habit of adding more trace calls throughout the code, makes it easy to see at runtime what is happening. E.g. in the case of this function can be nice to know...

- when we drained
- when we reused
- when we created

with some context params for each

etc....